wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/E2EINotificationActionsHandler.swift

```diff
@@ -106,40 +117,39 @@ final class E2EINotificationActionsHandler: E2EINotificationActions {
 
         let alert = await UIAlertController.reminderGetCertificate(timeLeft: formattedDuration) {
             Task {
-                await self.snoozeCertificateEnrollmentUseCase.invoke()
+                await self.snoozeCertificateEnrollmentUseCase.invoke(isUpdateMode: self.isUpdateMode)
+                self.isUpdateMode = false
             }
-            self.isUpdateMode = false
         }
         await targetVC.present(alert, animated: true)
     }
 
     // MARK: - Helpers
-    @MainActor
-    private func showGetCertificateErrorAlert(canCancel: Bool) {
-        let oauthUseCase = OAuthUseCase(rootViewController: targetVC)
-        let alert = UIAlertController.getCertificateFailed(canCancel: canCancel, isUpdateMode: isUpdateMode) {
-            Task { [weak self] in
-                guard let self else { return }
+
+    private func showGetCertificateErrorAlert(canCancel: Bool) async {
+        let oauthUseCase = OAuthUseCase(targetViewController: targetVC)
+        let alert = await UIAlertController.getCertificateFailed(canCancel: canCancel, isUpdateMode: isUpdateMode) {
+            Task {
                 do {
                     let certificateDetails = try await self.enrollCertificateUseCase.invoke(
                         authenticate: oauthUseCase.invoke)
                     await self.confirmSuccessfulEnrollment(certificateDetails)
                 } catch {
                     WireLogger.e2ei.error("failed to \(self.isUpdateMode ? "update" : "get") E2EI certification status: \(error)")
-                    isUpdateMode = false
                 }
+                self.isUpdateMode = false
             }
+        } cancelled: {[weak self] in
+            self?.isUpdateMode = false
```

Comment:
here we use weak self but not on the other closure , normal?

Reply:
self is captured inside a Task. Let me fix that for other closures in the file.

Reply:
done